EC-Switch-Seclog.py

```diff
@@ -0,0 +1,973 @@
+                'FailureToleranceCount': 9,
+                'MaxConcurrentCount': 10
+            }
+            cfn.create_stack_instances(
+                StackSetName=stackset, 
+                Regions=regions, 
+                Accounts=[account_id],
+                OperationPreferences=operationPreferences
+                )
+
+            time.sleep(5)
+            response = cfn.list_stack_set_operations(StackSetName=stackset)
+            while(any(x['Status'] == "RUNNING" for x in response['Summaries'])):
+                time.sleep(10)
+                response = cfn.list_stack_set_operations(StackSetName=stackset)
+            
+        
+            print(f"\033[2K\033[1GAdd stacks from StackSet {stackset} [{Status.OK.value}]")
+            
+        
+        except ClientError as error:
+            print(f"\033[2K\033[1GAdd stacks from StackSet {stackset} failed. Reason: {error.response['Error']['Message']} [{Status.FAIL.value}]")
+            print("Exiting...")
+            sys.exit(1)
+
+
+
+def update_stack(stack,session):
+    """
+    Function that updates a stack defined in the parameters
+        :stack:         The stack name
+        :template_data: dict holding CFT details
+        :params:        parameters to be passed to the stack
+        :return:        True or False
+    """
+    client = session.client('cloudformation')
+    
+    capabilities = []
+    tags = []
+    parameters = []
+    template_body = None
+    
+    print(f"Update {stack} stack ", end="")
+
+    try:
+        describe = client.describe_stacks(StackName=stack)
+        
+        if describe['Stacks'][0]['StackStatus'] not in ('CREATE_COMPLETE', 'UPDATE_COMPLETE','UPDATE_ROLLBACK_COMPLETE'):
+            print(f"Cannot update stack {stack}. Current status is : {describe['Stacks'][0]['StackStatus']} [{Status.FAIL.value}]")
+            print("Exiting...")
+            sys.exit(1)
+
+        if 'Capabilities' in describe['Stacks'][0]:
+            capabilities = describe['Stacks'][0]['Capabilities']
+
+        if 'Parameters' in describe['Stacks'][0]:
+            parameters = describe['Stacks'][0]['Parameters']
+
+        if 'Tags' in describe['Stacks'][0]:
+            tags =  describe['Stacks'][0]['Tags']
+
+
+        response = client.get_template(StackName=stack)
+        template_body = response['TemplateBody']
+
+    except ClientError as err:
+        print(f"\033[2K\033[1GUpdate Stack {stack} failed. Reason : {err.response['Error']['Message']} [{Status.FAIL.value}]")
+        print("Exiting...")
+        sys.exit(1)
+
+    print("in progress ", end="")
+    
+    with Spinner():
+        try:
+            
+            client.update_stack(
+                StackName=stack, 
+                TemplateBody=template_body, 
+                Parameters=parameters, 
+                Capabilities=capabilities, 
+                Tags=tags)
+            updated=False
+            while updated == False: 
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The `==` and `!=` operators use the compared objects&#39; ` __eq__` method to test if they are equal. To check if an object is a singleton, such as `None`, we recommend that you use the `is` identity comparison operator.

[Learn more](https://docs.python.org/3/reference/expressions.html#is)